src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -984,7 +984,7 @@ private long GetAssemblyId(WriteObjectInfo objectInfo)
             {
                 assemId = 0;
             }
-            else if (assemblyString.Equals(Converter.s_urtAssemblyString))
+            else if (assemblyString.Equals(Converter.s_urtAssemblyString) || assemblyString.Equals(Converter.s_urtAlternativeAssemblyString))
```

Comment:
This will bring it back to parity with the previous NETCore implementation.  This has two downsides:
1. If any type was added/moved into System.Private.CoreLib but wasn't in mscorlib on desktop, deserialization on desktop will fail.
2. Types which were in mscorlib on desktop but not in System.Private.CoreLib in .netcore will have a larger serialization payload.

Did you consider something like 
`Converter.s_urtAssembly.TryGetType(objectInfo.GetTypeFullName(), throwOnError:false) == objectInfo._objectType`

Could have a perf cost, which could be mitigated with a cache.  Potentially that probing could be done at the same time that objectInfo does other reflection.